### PR TITLE
replace cargo test with nextest

### PR DESCRIPTION
**Pull request by lingnoi**
Issues: #55 

# Definition of Done

The PR shall be merged **only if all of following items were checked.** In case an item is not applicable as described, please provide a short explanation right after the item in question. Separate the explanation by a semicolon and write it as bold text like **; requirements are handled in issue #xyz**:

- [x] documentation of all modules `/*/doc/swdesign` is up-to-date
- [x] requirements are up-to-date and requirements for new features have been added and mapped to source and test
- [x] conform to [unit verification strategy](https://eclipse-ankaios.github.io/ankaios/main/development/unit-verification/)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -7,18 +7,21 @@ on:
...
-      image: ghcr.io/eclipse-ankaios/devcontainer-base:0.6.2
+      image: ghcr.io/eclipse-ankaios/devcontainer-base:0.6.3
       options: --privileged
     steps:
       - uses: actions/checkout@v3.5.3
       - uses: Swatinem/rust-cache@v2.6.2
       
       - name: Run tests
-        run: RUST_LOG=debug cargo test -- --show-output --nocapture
+        run: RUST_LOG=debug cargo nextest run
```
Comment (inf17101) [inline .github/workflows/build.yml:24]: Does the new test exec lib include the prints/logs by default or why is "nocapture" removed now?

File: doc/docs/development/run-unit-tests.md [hunk 2]
```diff
@@ -14,14 +14,14 @@ This makes difficult to read trace logs, because it is hard to find which trace
 If you want to have traces sorted by unit tests, call this (recommended):
 
 ```shell
-cargo test -- --show-output
+cargo nextest -- --show-output
 ```
 
 Rust also allows to run only a subset of unit tests.
 You have to se the "filter string" in the command:
```
Comment (inf17101) [inline doc/docs/development/run-unit-tests.md:21]:
```suggestion
You have to set the "filter string" in the command:
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (lingnoi) [reply .github/workflows/build.yml:24]: You will get printf/logs outputs when only when a test failed, so you don't need to "pollute" the output with debug logs. Here an example output: 
image

Comment (inf17101) [reply .github/workflows/build.yml:24]: Ok I agree.

**Review by inf17101 — APPROVED** [review]
LGTM 👍

**Pull request merged**